packages/fuels-programs/src/logs.rs

```diff
@@ -34,26 +69,24 @@ impl LogDecoder {
             .iter()
             .extract_log_id_and_data()
             .filter_map(|(log_id, data)| {
-                self.type_lookup
+                self.log_formatters
                     .get(&log_id)
```

Comment:
We could. However, that scenario only happens when a script or contract is calling another contract. If a contract or script is decoding their own logs they will know the `log_id`.

If we would return an error here, that would force the user to abigen **all** external contracts that do any logging and maybe he wants to get longs from only some of them.

Reply:
Your comment makes sense. A possible solution would be to have some `LogResult` struct that holds both the correctly resolved logs and a list of errors. This way we can communicate a non-fatal failure and the user can investigate further if he wishes. wdyt?

Reply:
We've decided to move this into a [separate issue](https://github.com/FuelLabs/fuels-rs/issues/907) since the request isn't related to what this issue was about.

We can further discuss the idea there without blocking this PR.